hyperon_das/das.py

```diff
@@ -0,0 +1,569 @@
+import json
+from abc import ABC, abstractmethod
+from typing import Any, Dict, List, Optional, Set, Tuple, Union
+
+import requests
+from hyperon_das_atomdb import WILDCARD
+from hyperon_das_atomdb.adapters import InMemoryDB, RedisMongoDB
+from hyperon_das_atomdb.exceptions import (
+    AtomDoesNotExistException,
+    LinkDoesNotExistException,
+    NodeDoesNotExistException,
+)
+
+from hyperon_das.cache import AndEvaluator, LazyQueryEvaluator, ListIterator, QueryAnswerIterator
+from hyperon_das.client import FunctionsClient
+from hyperon_das.decorators import retry
+from hyperon_das.exceptions import (
+    InvalidDASParameters,
+    InvalidQueryEngine,
+    QueryParametersException,
+    UnexpectedQueryFormat,
+)
+from hyperon_das.logger import logger
+from hyperon_das.utils import Assignment, QueryAnswer
+
+
+class DistributedAtomSpace:
+    def __init__(self, **kwargs: Optional[Dict[str, Any]]) -> None:
+        atomdb_parameter = kwargs.get('atomdb', 'ram')
+        query_engine_parameter = kwargs.get('query_engine', 'local')
+
+        if atomdb_parameter == "ram":
+            self.backend = InMemoryDB()
+        elif atomdb_parameter == "redis_mongo":
+            mongo_hostname = kwargs.get('mongo_hostname')
```

Comment:
These parameters are not supposed to be unpacked here. We should just pass kwargs to MongoDB constructor and let it unpack any parameters it wants.